### PR TITLE
[fix][io] Close the kafka source connector got stuck

### PR DESCRIPTION
**Pull request by poorbarcode**
### Motivation

https://github.com/apache/pulsar/discussions/19880#discussioncomment-6026807 When Kafka connector is closing, it waits for the `runnerThread` to stop, but the task-close is running at the same thread, so it will be stuck.

### Modifications

run `close` in another thread.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: x

### CODE REVIEW

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java [hunk 1]
```diff
@@ -64,6 +66,7 @@ public abstract class KafkaAbstractSource<V> extends PushSource<V> {
     private volatile boolean running = false;
     private KafkaSourceConfig kafkaSourceConfig;
     private Thread runnerThread;
+    private final Executor executor = Executors.newSingleThreadExecutor();
```
Comment (codelipenghui) [inline pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java:69]: It's better to set a thread name for debugging purpose.

Comment (codelipenghui) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java:69]: And when the executor will be closed?

**Review by RobertIndie — APPROVED** [review]

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java (hunk 1 above)
Comment (mattisonchao) [inline pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java:69]: please close this executor. :)

Comment (poorbarcode) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java]: changed it to a static variable

Comment (poorbarcode) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java]: > It's better to set a thread name for debugging purpose.

fixed

> And when the executor will be closed?

changed it to a static variable

**Comment by mattisonchao** [comment]
We only use this thread once. Maybe we can new a thread and then run the close task?

**Review by codelipenghui — APPROVED** [review]

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java (hunk 1 above)
Comment (poorbarcode) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java]: already remove the executor, and just new one thread if needed

Comment (poorbarcode) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java]: already remove the executor, and just new one thread if needed

**Review by Demogorgon314 — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

**Comment by poorbarcode** [comment]
The failed [OWASP dependency check](https://github.com/apache/pulsar/actions/runs/5424405010/jobs/9865506786?pr=20698) will be fixed by https://github.com/apache/pulsar/pull/20699

**Pull request merged**